src/model/rpc.js

```diff
@@ -1,141 +1,162 @@
+    assert(typeof handlers[clpPacket.TYPE_FULFILL] === 'function', 'Fulfill handler missing')
+    assert(typeof handlers[clpPacket.TYPE_REJECT] === 'function', 'Reject handler missing')
+    assert(typeof handlers[clpPacket.TYPE_MESSAGE] === 'function', 'Message handler missing')
+    assert(typeof rpcUri === 'string', 'rpcUri must be string')
+    assert(typeof plugin === 'object', 'plugin must be provided')
+
     super()
     this._sockets = []
-    this._handlers = handlers
+    this.handlers = handlers
     this._rpcUri = rpcUri
-    this._account = account
-
-    assert(typeof this._handlers[Clp.TYPE_PREPARE] === 'function', 'Prepare handler missing')
-    assert(typeof this._handlers[Clp.TYPE_FULFILL] === 'function', 'Fulfill handler missing')
-    assert(typeof this._handlers[Clp.TYPE_REJECT] === 'function', 'Reject handler missing')
-    assert(typeof this._handlers[Clp.TYPE_MESSAGE] === 'function', 'Message handler missing')
+    this._plugin = plugin
+    this.debug = debug
   }
 
   addSocket (socket) {
+    _assertSocket(socket)
+    this.debug('adding socket')
     this._sockets.push(socket)
-    socket.on('message', this.handleMessage.bind(this, socket))
+    socket.on('message', async (message) => {
+      try {
+        await this.handleMessage(socket, message)
+      } catch (err) {
+        this.debug(`RPC Error: ${err.message}. Message was ${JSON.stringify(message)}`)
+      }
+    })
   }
 
   async handleMessage (socket, message) {
-    const { type, requestId } = Clp.readEnvelope(message)
-    let parsed
+    _assertSocket(socket)
+    const {type, requestId, data} = clpPacket.deserialize(message)
+    const typeString = clpPacket.typeToString(type)
+    if (data.transferId) {
+      data.id = data.transferId
+      delete data.transferId
+    }
 
+    this.debug(`received CLP packet (${typeString}, RequestId: ${requestId}): ${JSON.stringify(data)}`)
     switch (type) {
-      case Clp.TYPE_ACK:
-      case Clp.TYPE_RESPONSE:
-      case Clp.TYPE_ERROR:
-        this.emit('_' + requestId, message)
+      case clpPacket.TYPE_ACK:
+      case clpPacket.TYPE_RESPONSE:
+      case clpPacket.TYPE_ERROR:
+        this.emit('_' + requestId, type, data)
         return
 
-      case Clp.TYPE_PREPARE:
-        parsed = Clp.deserializePrepare(message)
-        break
-      case Clp.TYPE_FULFILL:
-        parsed = Clp.deserializeFulfill(message)
-        break
-      case Clp.TYPE_REJECT:
-        parsed = Clp.deserializeReject(message)
-        break
-      case Clp.TYPE_MESSAGE:
-        parsed = Clp.deserializeMessage(message)
+      case clpPacket.TYPE_PREPARE:
+      case clpPacket.TYPE_FULFILL:
+      case clpPacket.TYPE_REJECT:
+      case clpPacket.TYPE_MESSAGE:
         break
 
       default:
-        throw new Error(type + ' is not a valid Clp message type')
+        throw new Error(type + ' is not a valid clpPacket message type')
```

Comment:
This probably happened because of search-and-replace, but saying 'clpacket message' is a bit double.